lib/veritaserum.ex

```diff
@@ -27,44 +27,69 @@ defmodule Veritaserum do
       3
   """
   @spec analyze(String.t) :: Integer.t
-  def analyze(input) do
-    input
-    |> clean
-    |> String.split
-    |> split_on_emoticons
-    |> Enum.map(&String.trim/1)
-    |> analyze_list()
-    |> Enum.reduce(0, &(&1 + &2))
-  end
 
-  defp analyze_list([head | tail]) do
-    analyze_list(tail, head, [analyze_word(head)])
+  def analyze(input, options \\ [])
```

Comment:
With and without options we could split them into two functions, `analyze/1` and `analyze/2`, and pattern match the options.

To make this a bit cleaner we could extract the common parts to private functions and should be fine :)